### ISSUE TITLE
Question on posting file to media field directly

### ISSUE THREAD

**Issue by henrych4** [strapi/strapi#1585]
**Informations**
- **Node.js version**: 9.0.0
- **npm version**: 5.5.1
- **Strapi version**: 3.0.0-alpha.12.2
- **Database**: MongoDB
- **Operating system**: macOS High Sierra Version 10.13.1

**What is the current behavior?**
I have created a content type with field 'media'. And I would like to add file by sending post request and retrieve file location.
And from the response, I can see the following:
path: '/tmp/upload_1c8a06f4e1119d5ff82830d42936823b'
However I still cannot find the relevant file give the path.

From strapi website(https://strapi.io/documentation/guides/upload.html#examples), the workflow is to post file to 'Upload' and then link it back to corresponding model. But is it possible to directly post a file?

**Steps to reproduce the problem**
Create a content type called 'test' with media field called 'file' .
Add `console.log(ctx, ctx.request.body);` inside create action of controller of the content type.
Create a txt file called 'test.txt'
Run following command (you may need to enter your token, or you can set permission to avoid it):
```
curl -X POST -H 'Authorization: Bearer {token}' -F 'files=@test.txt' -F 'ref=test' -F 'field=files' http://localhost:1337/test
```

**What is the expected behavior?**
Return a correct path

**Suggested solutions**

**What is the expected behavior?**

**Comment by siffash** [comment]
@henrych4 hi henrych, I have the same issue here. Did you find any solution?

**Comment by henrych4** [comment]
I have seen your comment on another [post](https://github.com/strapi/strapi/issues/1316#issuecomment-416613667)

It seems there is something wrong with your token and thus you receive 403 response.

For upload issue, as mentioned [here](https://github.com/strapi/strapi/issues/1316#issuecomment-394806304), the normal way to upload file to a field is first upload it to `Upload` section and then link it to entity through admin panel.

However, there is another way by calling `uploadToEntity` method in `strapi-plugin-upload/services/Upload.js`.  [(source code)](https://github.com/strapi/strapi/blob/master/packages/strapi-plugin-upload/services/Upload.js#L141)
You need to provide `params` which is an object {model: 'ModelNameOfEntity', id: 'IDOfEntity'} and `file` (the file body).

**Comment by siffash** [comment]
Thank you @henrych4 
The problem was with uploading it to Upload section. I got errors 403 and 500.
To solve the 403 I needed to grant upload permissions in the Admin panel.
To solve the 500 I needed to create "uploads" directory in the "public" directory. Apparently heroku doesn't let strapi to create directories (since it worked only locally).
The question now is how to put the uploaded file inside a mongodb model and to remove it from the "uploads" directory. I'm very new to strapi. Is it possible to create some kind of middleware for the Upload plugin which will do this work?
